### ISSUE TITLE
Add a `ndf` command in Chebfun2

### ISSUE THREAD

**Issue by ajt60gaibb** [chebfun/chebfun#1842]
@bhashemi, Chebfun3 has the command `ndf`.  I think it is useful.  Why is it not called `numel`?  Do you think we should add the command to Chebfun2?

**Comment by aaustin141** [comment]
@bhashemi will have to give the definitive answer, but one reason why it's probably not named `numel` is that overriding `numel` would break our ability to create arrays of `chebfun3` objects, and that will break the current implementation of `chebfun3v`, which, like `chebfun2v`, stores the components in such an array.

I think we should try to avoid overriding `numel` if we can.  It's not really good practice; even [this page](https://uk.mathworks.com/help/matlab/matlab_oop/understanding-size-and-numel.html) from the MATLAB documentation cautions against it.  (It also cautions against overriding `size`, which we do, but I have a feeling that `numel` is somehow more fundamental...)

**Comment by bhashemi** [comment]
@ajt60gaibb you made two points: 
Firstly, about renaming `ndf` to `numel` I am happy with this to be changed as long as we agree about it. 
Here are two points: 
a) Thinking about continuous versus discrete one might argue that it is more appropriate for a `numel` in Chebfun to always create `Inf` in the output (similar to what we currently do in `size`). 
b) @aaustin141 notes that this renaming might be problematic. I am not completely sure how problematic it would be beyond forcing us to make some changes to `chebfun2v` and `chebfun3v`. 

Secondly, the question is what uses do we have in mind for `ndf` in `chebfun2`? There are two reasons for having `ndf` in `chebfun3`. 
c) One is to make comparisions with `chebfun3t` easier, as there is an obvious `ndf` in `chebfun3t` also. 
d) Another reason is that it is used e.g. in `minandmax3` as a rough estimate of how complicated a function is so that we can decide the size of a grid we use to sample for getting an initial guess. 

Point c) above applies to enabling comparisons between `chebfun2` and `chebfun2t` as in the following:
```
>> f2 = chebfun2(@(x,y) sin(10*(x+y))); r = length(f2); [m, n] = length(f2); 
ndf_f2 = r*(m+n)+r
f2t = chebfun3t(@(x,y,z) sin(10*(x+y))); ndf_f2t = ndf(f2t)
ndf_f2 =
   142
ndf_f2t =
        1225 
```
Are you thinking of using `ndf` for a different purpose? If you think this is cool enough, I'd be happy to write an `ndf` for `chebfun2`.

**Comment by trefethen** [comment]
A comment about overloading `size` -- this is a part of Chebfun that Cleve Moler gets a big kick out of, outputs like this:
```
>> size(A)
ans =
   Inf     3
```

**Comment by ajt60gaibb** [comment]
Yeah, I like the idea of an `ndf` command in Chebfun2.  Like you said, it would just be: 
```
function N = ndf( f ) 

r = length(f);
[m,n] = length(f);
N = r*(m+n)+r;   
end
```
We compare against tensor ndf by doing `[m,n]=length(f); m*n`.
